BuddyPress Installation Guide for WordPress Multisite (#275)
buddypress/buddypress: docs/user/installation-in-wordpress-multisite.md

# BuddyPress Installation Guide for WordPress Multisite

> Welcome to the BuddyPress installation guide for WordPress Multisite! This document is designed to help beginners install and configure BuddyPress on their WordPress Multisite network. Whether you're setting up a community site, a social network, or a collaborative space, BuddyPress is a powerful plugin that adds social networking features to your WordPress site.

Reviewer: ```suggestion
Welcome to the BuddyPress activation guide for WordPress Multisite! This document is designed to help beginners configure BuddyPress on their WordPress Multisite network. Whether you're setting up a community site, a social network, or a collaborative space, BuddyPress is a powerful plugin that adds social networking features to your WordPress site.
```

## Before You Begin

> 

Reviewer: ```suggestion

Just like there's only one common Users database table shared across your multisite network, BuddyPress only installs a common set a database tables for your network. This means, out of the box, BuddyPress does not provide a way to have more than one instance of it into your network so that you could have different communities for each of your network sites. In other words, activating BuddyPress network widely will only power your network with one BuddyPress instance you can choose to house on the main or a secondary site of your network.
```

#### Step 2: Network Activate BuddyPress

- From the Network Admin dashboard, go to **Plugins**.
- You will see BuddyPress listed among other installed plugins. Click on **Network Activate** to enable BuddyPress across your entire network.

#### Step 3: Configure BuddyPress

- In the Network Admin dashboard, navigate to **Settings → BuddyPress**.
- Configure the components you wish to enable network-wide, such as Friends, Groups, Private Messaging, etc.

### Network-wide Installation: BuddyPress Root Blog on a Secondary Site
One way to set up BuddyPress network-wide is to activate it across the entire network. Still, a secondary site should be designated as the main hub for all BuddyPress content and activities. This setup can be especially helpful if the network's main site has a specific function and is not meant to be used for community interactions.

#### Step 1: Identify Root Site and Modify Configuration

1. **Access Network Admin Dashboard**:
   - Navigate to `Dashboard → Network Admin`.

2. **Identify the Secondary Site**:
   - Click on the `Sites` link.
   - Locate and note the ID number of the secondary site you want to use as the root for BuddyPress.

> 

Reviewer: A screenshot like this one would be nice:

<img width="702" alt="network-admin-site-id" src="https://github.com/buddypress/buddypress/assets/1834524/0fda84d2-e0e2-4148-8d7a-4b50a9ea4e24">

3. **Configure wp-config.php**:
   - Access your server via FTP or your hosting file manager to open the `wp-config.php` file.

>    - Add the following line, replacing `$blog_id` with the actual ID of your chosen site:

Reviewer: ```suggestion
   - Add the following line, replacing `$blog_id` with the actual ID of your chosen site: in the above screenshot it's `4`.
```

     ```php

>      define('BP_ROOT_BLOG', $blog_id);

Reviewer: ```suggestion
     define('BP_ROOT_BLOG', 4);
```

     ```
   - Save and close the file.

> #### Step 2: Install and Activate BuddyPress

Reviewer: ```suggestion
#### Step 2: Network Activate BuddyPress
```

> 1. **Install BuddyPress**:
>    - Go back to the Network Admin dashboard.
>    - Navigate to `Plugins → Add New`.
>    - Search for "BuddyPress", then click `Install Now`.
> 
> 2. **Activate BuddyPress Network-Wide**:
>    - After installation, click `Activate`.
>    - Post activation, you will be directed to the BuddyPress Welcome screen, confirming the successful setup.

Reviewer: ```suggestion
- From the Network Admin dashboard, go to **Plugins**.
- You will see BuddyPress listed among other installed plugins. Click on **Network Activate** to enable BuddyPress across your entire network.
```

> #### Step 3: Configure BuddyPress for Multisite

Reviewer: ```suggestion
#### Step 3: Configure BuddyPress from the Network Admin
```

> 1. **Configure Network Settings**:
>    - In the Network Admin, navigate to `Settings → BuddyPress`.
>    - Set up and adjust the necessary settings and components, such as user profiles, groups, and site tracking, to suit your network’s needs.
>    - Confirm that all settings are properly saved and effectively applied network-wide, focusing on the newly designated root site.

Reviewer: ```suggestion
- In the Network Admin, navigate to `Settings → BuddyPress`.
- Set up and adjust the necessary settings and components, such as user profiles, groups, and site tracking, to suit your network’s needs.
- Confirm that all settings are properly saved and effectively applied network-wide, focusing on the newly designated root site.
```

Following these steps, you can successfully configure BuddyPress on a secondary site within your WordPress Multisite network, centralizing all BuddyPress activities on your chosen site and preserving the main site's primary functionalities.

### Setting Up BuddyPress on an Individual Site in WordPress Multisite

This guide provides detailed steps for configuring BuddyPress to operate on an individual site within a WordPress Multisite network. This allows for site-specific community features without affecting other sites in the network.

> #### Step 1: Install BuddyPress
> 
> 1. **Access Site Admin Dashboard**:
>    - Navigate to the site dashboard where you want to activate BuddyPress.
> 
> 2. **Install BuddyPress**:
>    - Go to `Plugins → Add New`.
>    - Search for "BuddyPress", then click `Install Now`.
>    - Do not activate it yet network-wide. Only proceed to activation in the individual site’s admin area.

Reviewer: This step should be explained into the [BuddyPress Installation Guide for Website Administrators](https://github.com/buddypress/buddypress/blob/master/docs/user/getting-started/activate-buddypress.md)

```suggestion

```

> #### Step 2: Activate BuddyPress on the Individual Site

Reviewer: ```suggestion
#### Step 2: Activate BuddyPress from the Individual Site's Dashboard
```

> 1. **Activate BuddyPress Locally**:
>    - In the same site's admin area where BuddyPress was installed, navigate to `Plugins`.
>    - Locate BuddyPress in the list and click `Activate` to enable it for this site.

Reviewer: ```suggestion
- In the same site's admin area, navigate to `Plugins`.
- Locate BuddyPress in the list and click `Activate` to enable it for this site **only**.
```

#### Step 3: Configure BuddyPress

> 1. **Set Up BuddyPress Components**:
>    - After activation, you'll find the BuddyPress settings under `Settings → BuddyPress` in the site’s admin panel.
>    - Configure the necessary components, such as Profiles, Groups, and Activity Streams. You may choose to activate only what's necessary for this site's community.

Reviewer: ```suggestion
- After activation, you'll find the BuddyPress settings under `Settings → BuddyPress` in the site’s admin panel.
- Configure the necessary components, such as Profiles, Groups, and Activity Streams. You may choose to activate only what's necessary for this site's community.
```
